src/project.c

```diff
@@ -1394,8 +1407,12 @@ char *xstrcpy(char **s1, const char *s2, const size_t n)
     }
 
     // See if size of destination string needs to grow
-    if (*s1) n1 = strlen(*s1);
-    if (n2 > n1) *s1 = realloc(*s1, (n2 + 1) * sizeof(char));
+    if (*s1 == NULL || n2 > strlen(*s1))
+    {
+        char* newPtr = realloc(*s1, (n2 + 1) * sizeof(char));
+        if (newPtr == NULL) return *s1;
+        *s1 = newPtr;
+    }
 
     // Copy the source string into the destination string
     strncpy(*s1, s2, n2+1);
```

Comment:
New lines 1410 to 1415 are unnecessary if we simply replace the original line 1401 with:
```
    if (*s1) strncpy(*s1, s2, n2+1);
```